services/src/main/java/org/keycloak/authentication/requiredactions/VerifyUserProfile.java

```diff
@@ -86,60 +85,24 @@ public void requiredActionChallenge(RequiredActionContext context) {
                 parameters = context.getHttpRequest().getDecodedFormParameters();
             }
 
-            context.challenge(createResponse(context, profile, parameters, errors));
-        }
-    }
-
-    @Override
-    public void processAction(RequiredActionContext context) {
-        EventBuilder event = context.getEvent();
-        event.event(EventType.VERIFY_PROFILE);
-        MultivaluedMap<String, String> formData = context.getHttpRequest().getDecodedFormParameters();
-
-        if (!context.getRealm().isEditUsernameAllowed()) {
-            formData.putSingle(UserModel.USERNAME, context.getUser().getUsername());
-        }
-
-        UserProfileProvider provider = context.getSession().getProvider(UserProfileProvider.class);
-        UserProfile profile = provider.create(UserProfileContext.UPDATE_PROFILE, formData, context.getUser());
-
-        try {
-            profile.update();
-            context.success();
-        } catch (ValidationException ve) {
-            List<FormMessage> errors = Validation.getFormErrorsFromValidation(ve.getErrors());
-            context.challenge(createResponse(context, profile, formData, errors));
+            context.challenge(createResponse(context, parameters, errors));
+            
+            EventBuilder event = context.getEvent().clone();
+            event.event(EventType.VERIFY_PROFILE);
+            event.detail("fields_to_update", collectFields(errors));
```

Comment:
Perhaps detail could be more clear that it fails when verifying the profile rather than update?

Reply:
I took it from the point of view of the admin who sees this event. This event means that verify profile found something what have to be improved in the user profile, and user is asked to update these fields. Maybe even event name isn't the best (I reused event you introduced), and this event should be something like `UPDATE_PROFILE_REQUESTED`.

But I'm not sure if there is any side effect of the event type removing/renaming.

BTW does every event type have to have `_ERROR` counterpart? It will nener be used for this event type (but VERIFY_PROFILE_ERROR exists)

Reply:
I like `UPDATE_PROFILE_REQUESTED`. Not sure though if we should change it now.

But yeah, `_ERROR` is used internally when the vent is related to an error.